`subscribe()` returns function to unsubscribe

Byron-TW opened the pull request:
This makes it even more compatible to the actual
api of `subscribe(...)` as documented here:

http://redux.js.org/docs/api/Store.html#subscribe

Commit:
`subscribe()` returns function to unsubscribe

This makes it even more compatible to the actual
api of `subscribe(...)` as documented here:

http://redux.js.org/docs/api/Store.html#subscribe

dmitry-zaets commented on test/index.js:
```diff
@@ -125,11 +125,29 @@ describe('redux-mock-store', () => {
+    });
+
+    unsubscribe()
+    store.dispatch(action);
```
This test contains not expectations. Would be good to test that subscribed callback was never called, by using https://github.com/mjackson/expect#spy-tonothavebeencalled

arnaudbenard commented on src/index.js:
```diff
@@ -49,7 +49,9 @@ export default function configureStore(middlewares=[]) {
           }
-          return null;
+          const index = listeners.indexOf(cb)
+          if (index < 0) { throw new Error('couldn\'t find callback in array') }
```
Why are you throwing an error here? 

In the official implementation, it doesn't throw an error. https://github.com/reactjs/redux/blob/b523aa4ae52f7afcbc66276cd18d5dc226c30b45/src/createStore.js#L101

Commit:
Don't throw if subscriber can't be found.

Commit:
Fix unsubscribe function

Previously it would not be idempotent, and generally mess up
the list of existing subscribers.

arnaudbenard commented on src/index.js:
```diff
@@ -49,7 +49,11 @@ export default function configureStore(middlewares=[]) {
+          return () => {
+            const index = listeners.indexOf(cb);
+            if (index < 0) { return; }
+            listeners.splice(index, 1)
```
Syntax correction

``` js
const index = listeners.indexOf(cb);

if (index < 0) { 
  return; 
}

listeners.splice(index, 1);
```

arnaudbenard commented on test/index.js:
```diff
@@ -125,11 +125,29 @@ describe('redux-mock-store', () => {
     store.dispatch(action);
   });
+
+  it('can unsubscribe subscribers', function (done) {
```
``` js
(done) => {
```

arnaudbenard commented on test/index.js:
```diff
@@ -125,11 +125,29 @@ describe('redux-mock-store', () => {
+      done(new Error('should never be called'));
+    });
+
+    unsubscribe()
```
missing `;`

Commit:
Syntax corrections
